### ISSUE TITLE
bug: incoming http POST queries are flagged as GET queries by the router

### ISSUE THREAD

**Issue by neslinesli93** [apollographql/router#865]
**Describe the bug**
After trying the `router` binary built from master, it looks like every operation performed on subgraphs uses a GET request. This is not a problem for queries, but it's a problem for mutations since our graphql server that implements a subgraph rejects mutation done with a GET rather than a POST. Maybe this started happening after the switch to axum?

**To Reproduce**
1. Perform a mutation to the router
2. The router returns 405 if the subgraph dealing with it has no support for mutation done with GET requests

**Expected behavior**
The router should work with both graphql queries and mutations

**Comment by Geal** [comment]
thanks for the report, this is a regression, we'll fix that soon

**Comment by BrynCooke** [comment]
This has been fixed and is part of https://github.com/apollographql/router/releases/tag/v0.1.0-preview.5, but leaving open as we have an issue with our unit tests.

**Comment by o0Ignition0o** [comment]
Ok this was a tricky one, but I've found and fixed it [this pull request](https://github.com/apollographql/router/pull/873).

We will thus merge it and cut a new release, hopefully tomorrow!

TLDR:
Incoming HTTP POST queries would be extracted as HTTP GET queries in our axum service handler, which would trigger the `forbid_http_get_mutations` plugin.
